libs/vscode/project-details/src/lib/pdv-state-machine.ts

```diff
@@ -0,0 +1,230 @@
+import type { PDVData } from '@nx-console/shared/types';
+import { assign, enqueueActions, fromPromise, setup } from 'xstate';
+
+export const machine = setup({
+  types: {
+    context: {} as Partial<PDVData> & {
+      multiSelectedProject: string | undefined;
+    },
+  },
+  actors: {
+    loadPDVData: fromPromise(async () => {
+      return {} as PDVData | undefined;
+    }),
+  },
+  actions: {
```

Comment:
Let's provide a comment to explain that these implementations will be provided by the consumer of the machine, either the view or tests. It's more clear if someone new is reading this.